src/js/components/common/localetime.js

```diff
@@ -0,0 +1,8 @@
+import React from 'react';
+import Time from 'react-time';
+import LocaleFormatString from './timeformat';
+
+export function LocaleTime(props) {
+  return <Time {...props} format={LocaleFormatString()} />;
+}
+export default LocaleTime;
```

Comment:
you could omit the reliance on the `timeformat` approach entirely if you took the selected format string straight from the redux store:

```suggestion
import { connect } from 'react-redux';
import { getUserSettings } from '../somewhere/are/the/selectors';

export const LocaleTime = ({ locale, stripTime, ...remainingProps }) => {
  const formatString = stripTime ? locale.dateFormat : `${locale.dateFormat} ${locale.timeFormat}`;
  return <Time {...remainingProps} format={formatString} />;
}

const mapStateToProps = state => {
  const { locale } = getUserSettings(state);
  return {
    locale
  };
};
export default connect(mapStateToProps)(LocaleTime);
```

you could of course also retrieve the `moment` localeData here and do the assembly based on this - but this way the component call could happen with the "proper" format already... I think it is rather unlikely that the constant `localeData` will change between selection and rendering, but the way you changed the call to `moment` now I'm fine with both approaches 👍